src/Domain/Services/CreateAccount.php

```diff
@@ -116,11 +116,10 @@ public function from_order_request( \WC_Order $order, \WP_REST_Request $request
 		}
 
 		$customer_id = $this->create_customer_account(
-			$order->get_billing_email(),
-			$order->get_billing_first_name(),
-			$order->get_billing_last_name()
+			$request['billing_address']['email'],
+			$request['billing_address']['first_name'],
+			$request['billing_address']['last_name']
```

Comment:
The `$order` param is now only used once in `from_order_request()` to set the user (`$order->set_customer_id`). You could factor that out to `get_route_post_response()` and then we wouldn't need to pass $order to `from_order_request`, that might simplify this more.

Reply:
(Not a blocker, just a thought)